.github/CONTRIBUTING.md

```diff
@@ -1,121 +1,120 @@
+## Run tests (offline)
 
-Example:
+If the calls to the Sheets API have not changed, you can run the tests offline. Otherwise, you will have to [run them online](#run-tests-online) to record the new API calls.
 
-```
-tox -e py -- -k test_find -v -s
+This will use the currently recorded HTTP requests + responses. It does not make any HTTP calls, and does not require an active internet connection.
+
+```bash
+tox -e py
 ```
 
-**Note:** gspread uses [vcrpy](https://github.com/kevin1024/vcrpy) to record and replay HTTP interactions with Sheets API.
+### Run a specific test
 
-You must in that case provide a service account credentials in order to make the real HTTP requests, using `GS_CREDS_FILENAME` environment variable.
+```bash
+tox -e py -- -k TEST_NAME -v -s
+```
 
-You can control vcrpy's [Record Mode](https://vcrpy.readthedocs.io/en/latest/usage.html#record-modes) using `GS_RECORD_MODE` environment variable.
+## Format
 
-The following command will run the entire test suite and record every HTTP request.
-```
-GS_RECORD_MODE=all GS_CREDS_FILENAME=<YOUR_CREDS.json> tox -e py
+```bash
+tox -e format
 ```
 
-You need to update the recorded HTTP requests in the following cases:
+## Lint
 
-- new test is added
-- a existing test is updated and does a new HTTP request
-- gspread is updated and does a new HTTP request
+```bash
+tox -e lint
+```
 
-In any of the above cases:
+## Render Documentation
 
-- remove the file holding the init/teardown of the test suite.
+The documentation uses [reStructuredText](http://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html#rst-index) markup and is rendered by [Sphinx](http://www.sphinx-doc.org/).
 
-  ex: for the file `tests/cell_test.py` delete `tests/cassettes/CellTest.json`
-- please update the HTTP recording using the command above
-- set the `GS_RECORD_MODE` to `new_episodes`.
+```bash
+tox -e doc
+```
 
-This will tell `vcrpy` to record only new episodes and replay existing episodes.
+The rendered documentation is placed into `docs/build/html`. `index.html` is an entry point.
 
-**Note:** this will mostly result in a lot of udpated files under `tests/cassettes/` don't forget to add them in your PR.
+## Run tests (online)
 
-Add these new files a dedicated commit, in order to make the review process easier please.
+gspread uses [vcrpy](https://github.com/kevin1024/vcrpy) to record and replay HTTP interactions with Sheets API.
 
-The following command will replay existing requests and record new requests:
-```
-GS_RECORD_MODE=new_episodes GS_CREDS_FILENAME=<YOUR_CREDS.json> tox -e py
-```
+### `GS_CREDS_FILENAME` environment variable
 
-Then run the tests in offline mode to make sure you have recorded everything.
+You must provide service account credentials using the `GS_CREDS_FILENAME` environment variable in order to make HTTP requests to the Sheets API.
 
-```
-tox -e py
-```
+[Obtain OAuth2 credentials from Google Developers Console](http://gspread.readthedocs.org/en/latest/oauth2.html).
```

Comment:
In the record mode we expect a service account in order to run the test onlie.

The correct link is: `https://docs.gspread.org/en/latest/oauth2.html#for-bots-using-service-account`